File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-5.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 0
        r = 1

        while r < len(nums):
            if nums[r] <= nums[l]:
                r += 1
                continue
            l += 1

            nums[r], nums[l] = nums[l], nums[r]
            
        return l + 1

File: Data Structures & Algorithms/two-integer-sum/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const numsMap = {}

        for (let i = 0, n = nums.length; i < n; i++){
            numsMap[nums[i]] = i;
        }

        for (let i = 0, n = nums.length; i < n; i++){
            const match = target - nums[i];
            if (match in numsMap && i !== numsMap[match]) {
                return [i, numsMap[match]];
            }
        }

    }
}
